src/masternodeman.cpp

```diff
@@ -333,7 +324,7 @@ CMasternode* CMasternodeMan::GetNextMasternodeInQueueForPayment(int nBlockHeight
         //make sure it has as many confirmations as there are masternodes
         if(mn.GetCollateralAge() < nMnCount) continue;
 
-        vecMasternodeLastPaid.push_back(std::make_pair(mn.GetLastPaidBlock(), mn.vin));
+        vecMasternodeLastPaid.push_back(std::make_pair(mn.GetLastPaidBlock(), &mn));
```

Comment:
With your changes in the loop below, the result should be independent of the order of this vector.  So you shouldn't need to sort the vector at all.

But maybe there is some significance to the Find that you replaced ?  I don't fully understand what that was supposed to be for.